internal/blueprint/repository_customizations_test.go

```diff
@@ -93,3 +96,340 @@ func TestCustomRepoSerialization(t *testing.T) {
+				"example-1.repo": {
+					{
+						Id:       "example-1",
+						BaseURLs: []string{"http://example-1.com"},
+						GPGKeys:  []string{"http://example-1.com/gpgkey"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+				"example-2.repo": {
+					{
+						Id:       "example-2",
+						BaseURLs: []string{"http://example-2.com"},
+						GPGKeys:  []string{"http://example-2.com/gpgkey"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+			},
+		},
+		{
+			Name: "Test inline gpgkeys",
+			Repos: []RepositoryCustomization{
+				{
+					Id:        "example-1",
+					BaseURLs:  []string{"http://example-1.com"},
+					GPGKeys:   []string{"fake-gpg-key-1"},
+					GPGCheck:  common.ToPtr(true),
+					SSLVerify: true,
+				},
+				{
+					Id:        "example-2",
+					BaseURLs:  []string{"http://example-2.com"},
+					GPGKeys:   []string{"fake-gpg-key-2"},
+					GPGCheck:  common.ToPtr(true),
+					SSLVerify: true,
+				},
+			},
+			WantRepoConfig: map[string][]rpmmd.RepoConfig{
+				"example-1.repo": {
+					{
+						Id:       "example-1",
+						BaseURLs: []string{"http://example-1.com"},
+						GPGKeys:  []string{"file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-1-0"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+				"example-2.repo": {
+					{
+						Id:       "example-2",
+						BaseURLs: []string{"http://example-2.com"},
+						GPGKeys:  []string{"file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-2-0"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+			},
+		},
+		{
+			Name: "Test multiple inline gpgkeys",
+			Repos: []RepositoryCustomization{
+				{
+					Id:        "example-1",
+					BaseURLs:  []string{"http://example-1.com"},
+					GPGKeys:   []string{"fake-gpg-key-1", "fake-gpg-key-2"},
+					GPGCheck:  common.ToPtr(true),
+					SSLVerify: true,
+				},
+				{
+					Id:        "example-2",
+					BaseURLs:  []string{"http://example-2.com"},
+					GPGKeys:   []string{"fake-gpg-key-1", "fake-gpg-key-2"},
+					GPGCheck:  common.ToPtr(true),
+					SSLVerify: true,
+				},
+			},
+			WantRepoConfig: map[string][]rpmmd.RepoConfig{
+				"example-1.repo": {
+					{
+						Id:       "example-1",
+						BaseURLs: []string{"http://example-1.com"},
+						GPGKeys:  []string{"file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-1-0", "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-1-1"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+				"example-2.repo": {
+					{
+						Id:       "example-2",
+						BaseURLs: []string{"http://example-2.com"},
+						GPGKeys:  []string{"file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-2-0", "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example-2-1"},
+						GPGCheck: common.ToPtr(true),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.Name, func(t *testing.T) {
+			got, _, err := RepoCustomizationsToRepoConfigAndGPGKeyFiles(tt.Repos)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.WantRepoConfig, got)
+		})
+	}
+}
+
+func TestRepoCustomizationsToGPGKeyFiles(t *testing.T) {
```

Comment:
Functionally, this is all good. Logically, I think that you could test conversion of repo customizations to `RepoConfig` and `File` in a single test case. However I'm OK with keeping these test cases split.